src/main/java/net/dries007/tfc/objects/entity/ai/EntityAITamableAvoidPlayer.java

```diff
@@ -0,0 +1,29 @@
+package net.dries007.tfc.objects.entity.ai;
+
+import net.minecraft.entity.ai.EntityAIAvoidEntity;
+import net.minecraft.entity.passive.EntityAnimal;
+
+import net.dries007.tfc.api.types.IAnimalTFC;
+
+//AI to avoid players unless familiarized. Must be applied to EntityAnimal that implements IAnimalTFC
+public class EntityAITamableAvoidPlayer extends EntityAIAvoidEntity
+{
+    public EntityAITamableAvoidPlayer(EntityAnimal entityIn, Class classToAvoidIn, float avoidDistanceIn, double farSpeedIn, double nearSpeedIn)
+    {
+        super(entityIn, classToAvoidIn, avoidDistanceIn, farSpeedIn, nearSpeedIn);
+        if (!(entityIn instanceof IAnimalTFC))
+        {
+            throw new AssertionError();
+        }
+    }
```

Comment:
```suggestion
public class EntityAITamableAvoidPlayer<T extends EntityCreature & IAnimalTFC> extends EntityAIAvoidEntity<EntityPlayer>
{
    public EntityAITamableAvoidPlayer(T entityIn, float avoidDistanceIn, double farSpeedIn, double nearSpeedIn)
    {
        super(entityIn, EntityPlayer.class, avoidDistanceIn, farSpeedIn, nearSpeedIn);
    }
```

Reply:
Done. Also needed to consider begging in wolf avoidance.